### ISSUE TITLE
Latest builds break SelfLights

### ISSUE THREAD

**Issue by Trava715** [ac-custom-shaders-patch/acc-extension-config#141]
![image](https://user-images.githubusercontent.com/43389722/138169532-50b7b13d-963f-48b5-8f8d-c2c19da741c8.png)
![image](https://user-images.githubusercontent.com/43389722/138169594-8082faa0-8ce9-4ac7-9a59-24f7061224d8.png)

Happens on 1.76 preview 134 and preview 96
Seems to be related to high COLOR values (1000+)

**Comment by Trava715** [comment]
Still not fixed as of CSP 1.76
https://acstuff.ru/u/comparison/P_f

**Comment by gro-ove** [comment]
Not entirely sure if it’s a bug if COLOR is above 1000. For comparison, this light is `1, 1, 1, 30`:

![image](https://user-images.githubusercontent.com/3996502/152254257-348f9bf9-d13f-44df-8c29-7014603d4d96.png)

There was a bug earlier where in some conditions overlapping regular emissives and custom emissives would result in emissives being extra dim which broke older cars. Apparently fix for that, while solved issues for those older cars, now broke some newer cars which as a way to fight that bug used extremely high brightness values. Not entirely sure what to do now, to be honest.

As one possible solution, I could clamp COLOR by something like 50, but I worry it might break some others cars… Maybe I’ll come up with a better solution a bit later. 🤔

**Comment by Trava715** [comment]
```ini
[EMISSIVE_LIGHT_...]
NAME=EXT_HEADLIGHT_GLASS?_?
COLOR =6,5,4, 1
CHANNEL = 2

[LIGHT_EXTRA_...]
BOUND_TO = EXT_HEADLIGHT_GLASS_R:2
COLOR=0.9, 0.871429, 0.807143, 6600
```

The extra light is not overlapping anything, it's directly over carpaint.

**Comment by gro-ove** [comment]
When config binds light to an original emissive, it doesn’t take ksEmissive directly, instead it calculates brightness multiplier to use with light’s COLOR. To calculate that multiplier light takes ksEmissive, divides it by BOUND_EMISSIVE_MAX and then raises result to power of BOUND_EXP. Default value for BOUND_EMISSIVE_MAX is 500, default value for BOUND_EXP is 1. So if you’re binding light to a small emissive set in lights.ini with something like `LIGHT=5, 5, 5`, brightness multiplier for light when emissive is active will be 5/500=1%. It’s not the best system, but originally it was created to work with headlights which could occasionally flash, like flashing in pits, and when doing so could have strange emissive values. Simply taking maximum possible value and using that as default value for BOUND_EMISSIVE_MAX would mess up quite a few Kunos cars. And that’s why in such cases crazy high COLOR values can compensate for that.

Originally custom emissives worked differently: there, CSP knows maximum brightness they can reach, so why not use that for multiplier. However, some time ago with another optimization rewrite I broke something and they reverted to using BOUND_EMISSIVE_MAX. Some people started to report cars with older configs losing extra lights. With 0.1.75 I found the bug and fixed it, but that inadvertently broke cars made while bug was still present. 

With 0.1.77 I added a fix. If half of extra lights are brighter than 400, or at least two are brighter than 1600 a fix will be applied to all lights reverting them to broken behaviour. Configs can use `[BASIC] NO_NEED_FOR_LIGHTS_FIX=1` to cancel out this behaviour.

Seems like it’s not enough though, so I’ll change it that only a single light would need to be brighter than 1600, or it would look for a `[BASIC] NEEDS_LIGHTS_FIX=1` flag. Hopefully that would solve the problem instead of creating new problems.

Sorry about that, it’s my fault for not documenting things properly. That thing with saving guessed configs… Going to work more on that wiki soon.
